PaymentMethods/HeidelpayAbstractPaymentMethod.php

```diff
@@ -821,4 +822,18 @@ public function setAsync()
             $frontend['RESPONSE_URL']              // Response url from your application
         );
     }
+
+    /**
+     * @param $transactionID string
+     * @return bool
+     */
+    public function heidelpayTransactionExists($transactionID)
+    {
+        $collection = $this->transactionCollectionFactory->create();
+
+        /** @var \Heidelpay\Gateway\Model\Transaction $heidelpayTransaction */
+        $heidelpayTransaction = $collection->loadByTransactionId($transactionID);
+
+        return !$heidelpayTransaction === null && !$heidelpayTransaction->isEmpty();
```

Comment:
Evtl. n bisschen einfacher, oder?

```suggestion
        return $heidelpayTransaction instanceof Transaction;
```